### PR TITLE
Add subdirectories to be part of the task prefix

### LINKED ISSUE #4066: task_name_prefix doesnt work with folders
##### Summary
name[prefix] doesnt work with tasks in folders:

We have a Role with the following structure:

roles/haproxy/
├── defaults
├── files
├── handlers
├── tasks
│   ├── copy_certificates.yml
│   ├── haproxy1
│   │   ├── config.yml
│   │   └── main.yml
│   ├── haproxy2
│   │   ├── confd.yml
│   │   ├── config.yml
│   │   └── main.yml
│   ├── main.yml
│   └── set_backend_state.yml
└── templates

The tasks are named like: Folder | File | Taskname

##### Issue Type
name[prefix]: Task name should start with 'config | '.
roles/haproxy/tasks/haproxy2/config.yml:2 Task/Handler: HAproxy2 | config | Create directories

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 24.2.0 using ansible-core:2.16.2 ansible-compat:4.1.11 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.7
```

- ansible installation method: OS package
- ansible-lint installation method:  OS package

#### Desired Behavior
task_name_prefix with optional folder name?

### PR DESCRIPTION
**Pull request by cavcrosby**
Fixes #4066.

### CODE REVIEW

File: src/ansiblelint/rules/name.py [hunk 1]
```diff
@@ -125,10 +128,15 @@ def _check_name(
...
-            if lintable.kind == "tasks" and lintable.path.stem != "main":
+            full_stem = self._find_full_stem(lintable)
+            stems = [
+                self._collection.options.task_name_prefix.format(stem=stem)
+                for stem in wcmatch.pathlib.PurePath(
+                    full_stem,
+                ).parts
+            ]
+            prefix = "".join(stems)
+            if lintable.kind == "tasks" and full_stem != "main":
```
Comment (cavcrosby) [inline src/ansiblelint/rules/name.py:139]: I would think any task file embedded within a subdirectory should be required to prefix each task with its stems. Even if the file was named `main.yml`.

File: src/ansiblelint/rules/name.md [hunk 2]
```diff
@@ -21,12 +21,21 @@ If you want to ignore some of the messages above, you can add any of them to the
 
 ## name[prefix]
 
-This rule applies only to included task files that are not named `main.yml`. It
-suggests adding the stem of the file as a prefix to the task name.
+This rule applies only to included task files that are not named `main.yml` or
```
Comment (cavcrosby) [inline src/ansiblelint/rules/name.md:24]: I do think this sentence can be a tad confusing until the next two paragraphs are read. I am open to suggestions here.

Comment (ssbarnea) [reply src/ansiblelint/rules/name.md:24]: @cidrblock WDYT?

I personally do not have a strong opinion here, but I kinda like the idea of not forcing people to add `| main |` extra prefix for `tasks/mains.yml` which is like 95% of them.

**Review by ansibuddy — COMMENTED** [review]
+1

**Pull request merged**